packages/rpc-provider/src/ws/Provider.ts

```diff
@@ -138,7 +141,9 @@ export default class WsProvider implements WSProviderInterface {
 
       const WS = await getWSClass();
 
-      this.#websocket = new WS(this.#endpoints[this.#endpointIndex]);
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore - WS may be an instance of w3cwebsocket, which supports headers
```

Comment:
There is likely a more elegant way to do this - but I wasn't sure which approach was preferred. Happy to update in that case

Reply:
So this is where not doing this 100% in `getWSClass` bytes us. So generally, it would be -

```js
function getWSClass (): typeof WebSocket | typeof W3CWebSocket { ...} 

...

this.websocket = WS instance WebSocket
  ? new WS(...)
  : new WS(...)
```

Not the end of the world (although we know TS-overrides a a bit smelly), but the imp[act otherwise if probably much worse. So the comment here is probably the lesser of all evils and should rather be left as-is until the typing of `getWsClass` is done properly. (And not hacked about via `as unknown as ...` - which created the above)

TL;DR Certainly not perfect, but follows the lead of an issue introduced elsewhere - so with the comment, should be all good.